### DISCUSSION TITLE
Text `pattern()`- nullable and empty string suport

### DISCUSSION THREAD

**Discussion by shivanshmehta3** [instancio/instancio#798] [Ideas]
`gen.text().pattern("#d#d")` should have a support of generating null and/or empty text strings.

**Comment by armandino** [comment]
@shivanshmehta3 thanks for the suggestion. Will be added in the next release.

**Comment by armandino** [comment]
@shivanshmehta3 - please upgrade to `3.4.2`, which adds `nullable()` and `allowEmpty()` methods, e.g.

```java
gen.text().pattern("#d#d").nullable().allowEmpty()
```

**Reply by shivanshmehta3** [reply]
sure thanks for the quick update.
